nf_core/components/snapshot_generator.py

```diff
@@ -0,0 +1,174 @@
+        if nftest_err:
+            syntax = Syntax(nftest_err.decode(), "java", theme="ansi_dark")
+            print(Panel(syntax, title="nf-test error"))
+
+    def generate_snapshot(self) -> bool:
+        """Generate the nf-test snapshot using `nf-test test` command
+
+        returns True if the test was successful, False otherwise
+        """
+
+        log.debug("Running nf-test test")
+
+        # set verbose flag if self.verbose is True
+        verbose = "--verbose --debug" if self.verbose else ""
+        update = "--update-snapshot" if self.update else ""
+        self.update = False  # reset self.update to False to test if the new snapshot is stable
+
+        result = nf_core.utils.run_cmd(
+            "nf-test",
+            f"test --tag {self.component_name} --profile {os.environ['PROFILE']} {verbose} {update}",
+        )
+        if result is not None:
+            nftest_out, nftest_err = result
+            self.display_nftest_output(nftest_out, nftest_err)
+            # check if nftest_out contains obsolete snapshots
+            pattern = r"Snapshot Summary:.*?(\d+)\s+obsolete"
+            compiled_pattern = re.compile(pattern, re.DOTALL)  # re.DOTALL to allow . to match newlines
+            obsolete_snapshots = compiled_pattern.search(nftest_out.decode())
+            if obsolete_snapshots:
+                self.obsolete_snapshots = True
+
+            # check if nf-test was successful
+            if "Assertion failed:" in nftest_out.decode():
+                log.error("nf-test failed")
+                return False
+            else:
+                log.debug("nf-test successful")
+                return True
+        else:
+            log.error("nf-test failed")
+            return False
+
+    def check_snapshot_stability(self) -> bool:
+        """Run the nf-test twice and check if the snapshot changes"""
+        log.info("Generating nf-test snapshot")
+        if not self.generate_snapshot():
+            return False  # stop here if the first run failed
+        log.info("Generating nf-test snapshot again to check stability")
+        if not self.generate_snapshot():
+            log.error("nf-test snapshot is not stable")
+            return False
+        else:
+            if self.obsolete_snapshots:
+                # ask if the user wants to remove obsolete snapshots using nf-test --clean-snapshot
+                if self.no_prompts:
+                    log.info("Removing obsolete snapshots")
+                    nf_core.utils.run_cmd("nf-test", "clean-snapshot")
+                else:
+                    answer = Confirm.ask("nf-test found obsolete snapshots. Do you want to remove them?", default=True)
+                    if answer:
+                        log.info("Removing obsolete snapshots")
+                        nf_core.utils.run_cmd(
+                            "nf-test",
+                            f"test --tag {self.component_name} --profile {os.environ['PROFILE']} --clean-snapshot",
```

Comment:
Why is this command different from the one in line [163](https://github.com/nf-core/tools/pull/2490/files#diff-6a3da60e503ad23420f49b464256a5f118145f986cd19df97ddcc9ca25daed10R163)?

Reply:
yes, sorry, 163 should also have the `--tag` etc